[Stepper] Always pass state props to connector

baterson opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/master/CONTRIBUTING.md#sending-a-pull-request).  

Hello! This is a fix for https://github.com/mui-org/material-ui/issues/21325  
@oliviertassinari I followed your suggestion on fixing this issue. But I had to rewrite a couple of tests to make them green again.

Closes #21325

oliviertassinari commented on packages/material-ui/src/Stepper/Stepper.test.js:
```diff
@@ -176,6 +173,25 @@ describe('<Stepper />', () => {
+        </Stepper>,
+      );
+
+      expect(wrapper.find(StepConnector).at(0).props().active).to.equal(true);
```
Please test this with the CSS API, also, could you use testing library instead of enzyme for the next tests?

baterson replied:
Sure, I'll do it tonight!

oliviertassinari replied:
Already fixed :)

baterson replied:
@oliviertassinari Thanks! Are you planning to rewrite all existing tests to testing library? It looks like it's a good possibility for future contributions :)

oliviertassinari replied:
Yes, we are rewriting existing tests to testing-library. It would make a great future contributions :)

baterson commented:
@eps1lon @oliviertassinari Thanks for the review! About test updates - I updated only those tests which stopped working after changes. Some of them were an easy fix - changing `shallow` to `mount`  rendering. 
But this one https://github.com/mui-org/material-ui/pull/21370/files#diff-1123958d59e83fcef53e811ac6d6dae6R49 
required to change assertions and components because `Stepper` was no longer rendering `StepConnector` and `Step` in one Array, so the number of children was changed.  
Also, the `Circle` svg component was causing warnings and test crush, so I changed it to.
https://github.com/mui-org/material-ui/pull/21370/files#diff-1123958d59e83fcef53e811ac6d6dae6L137
